classes/SnippetLoader.php

```diff
@@ -0,0 +1,166 @@
+        }
+
+        return $partialSnippetMap[$snippetCode];
+    }
+
+    /**
+     * Save to the cache the component snippets loaded for this page.
+     * Should be called once after all snippets are loaded to avoid multiple serializations.
+     */
+    public static function saveCachedSnippets(CmsPage $page): void
+    {
+        if (empty(static::$pageSnippetsCache)) {
+            return;
+        }
+
+        Cache::put(
+            static::getMapCacheKey($page),
+            serialize(static::$pageSnippetsCache),
+            Carbon::now()->addDay()
+        );
+    }
+
+    /**
+     * Register back to the current controller all component snippets previously saved.
+     * This make AJAX handlers of these components available.
+     */
+    public static function restoreComponentSnippetsFromCache(CmsController $controller, CmsPage $page): void
+    {
+        $componentSnippets = static::fetchCachedSnippets($page);
+
+        foreach ($componentSnippets as $componentInfo) {
+            try {
+                static::attachComponentSnippetToController($componentInfo, $controller);
+            } catch (\Exception $e) {
+            }
+        }
+    }
+
+    /**
+     * Attach a component-based snippet to a controller.
+     *
+     * Register the component if it is not defined yet.
+     * This is required because not all component snippets are registered as components,
+     * but it's safe to register them in render-time.
+     *
+     * If asked, the run lifecycle events of the component can be run. This is required for
+     * component that are added late in the page execution like with the twig filter.
+     *
+     * @throws SystemException if component code is reserved.
+     */
+    protected static function attachComponentSnippetToController(array $componentInfo, CmsController $controller, bool $triggerRunEvents = false): void
+    {
+        $componentManager = ComponentManager::instance();
+
+        if (!$componentManager->hasComponent($componentInfo['component'])) {
+            $componentManager->registerComponent($componentInfo['component'], $componentInfo['code']);
+        }
+
+        $component = $controller->addComponent(
+            $componentInfo['component'],
+            $componentInfo['code'],
+            $componentInfo['properties']
+        );
+
+        if ($triggerRunEvents) {
+            if ($component->fireEvent('component.beforeRun', [], true)) {
+                return;
+            }
+
+            if ($component->onRun()) {
+                return;
+            }
+
+            if ($component->fireEvent('component.run', [], true)) {
+                return;
+            }
+        }
+    }
+
+    /**
+     * Store a component snippet to the cache.
+     * The cache is not actually saved; saveCachedSnippets() must be called to persist the cache.
+     */
+    protected static function cacheSnippet(string $alias, array $snippetInfo): void
```

Comment:
```suggestion
    protected static function cacheSnippet(array $snippetInfo): void
```

Reply:
I think there's no need to pass a seperate variable for the alias, what do you think?